src/argus/notificationprofile/media/__init__.py

```
@@ -23,29 +25,19 @@
     "get_notification_media",
 ]
 
-
-# Poor mans's plugins
-MEDIA_CLASSES = {
-    NotificationProfile.Media.EMAIL: getattr(
-        settings, "DEFAULT_EMAIL_MEDIA", "argus.notificationprofile.media.email.EmailNotification"
-    ),
-    NotificationProfile.Media.SMS: getattr(settings, "DEFAULT_SMS_MEDIA", None),
-}
-
-
+# Import media classes
 def _import_class_from_dotted_path(dotted_path: str):
     module_name, class_name = dotted_path.rsplit(".", 1)
     module = importlib.import_module(module_name)
     class_ = getattr(module, class_name)
     return class_
 
 
-for media_type, media_class in MEDIA_CLASSES.items():
-    if not media_class or not isinstance(media_class, str):
-        continue
-    # Dotted paths here!
-    # TODO: Raise Incident if media_class not importable
-    MEDIA_CLASSES[media_type] = _import_class_from_dotted_path(media_class)
+# TODO: Raise Incident if media_class not importable?
+MEDIA_PATHS = getattr(settings, "MEDIA_PATHS", None)
+if not MEDIA_PATHS:
+    LOG.error("Notification: Apps: MEDIA_PATHS not set, cannot configure media")
+MEDIA_CLASSES = [_import_class_from_dotted_path(media_path) for media_path in MEDIA_PATHS]
```

You could just raise an exception, see what Django does. ImportError, settings not configured error?

(Do we want to allow argus to be running with no way to send any notifications? Not a rhetorical question!)

---

I think it should not raise an exception there, but probably rather when we try to send notifications, but have nowhere to send it to?

---

See my parenthetical: if we allow Argus to run with zero notification media set up, then the correct place to raise an error is when sending a notification. If Argus MUST have at least one notification medium set up then we must flag that none are to be found as early as possible, as we then have a malformed settings-file/broken deployment. So it depends, as usual. Still, we could flag it as an error as early as possible if a destination-config has a medium that is not supported (because we cannot find the library for it).

---

I think the more sensible thing is to require Argus to have at least one notification medium set up 

And I will have a look that we flag it as an error if one wants to set a destination-config that is not supported